### PR TITLE
HPCC-27460 Fix invalid index error

### PR DESCRIPTION
**Pull request by gfortil**
This error occurs when users try to import their own virtual network. The intended behavior was to not fetch data from the incorporated virtual network module once the virtual_network block is set. However, the current behavior is that Terraform tries to fetch even when it wouldn't use that data from the incorporated module. The issue is due to the use of try and catch. The correct implementation should use the can() function instead. To successfully test this, the host machine must not have python installed and the tester needs to import their own virtual network.

Signed-off-by: Godson Fortil <godson.fortil@lexisnexisrisk.com>

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -12,7 +12,7 @@ locals {
     var.metadata.resource_group_type != "" ? { resource_group_type = var.metadata.resource_group_type } : {}
   ) : module.metadata.names
   tags            = var.disable_naming_conventions ? merge(var.tags, { "admin" = var.admin.name, "email" = var.admin.email }) : merge(module.metadata.tags, { "admin" = var.admin.name, "email" = var.admin.email }, try(var.tags))
-  virtual_network = var.virtual_network != null ? var.virtual_network : data.external.vnet[0].result
+  virtual_network = can(var.virtual_network.private_subnet_id) && can(var.virtual_network.public_subnet_id) && can(var.virtual_network.route_table_id) ? var.virtual_network : data.external.vnet[0].result
```
Comment (stuartort) [inline locals.tf:15]: Should all of then "can" calls be placed in inside parens?  Seems as if you want all 3 conditions to be true for the expression to evaluate to true.

Question... Due to limited terraform knowledge, is "can" better / more effecient than testing != null

Comment (gfortil) [reply locals.tf:15]: Can only supports a single argument. Can is cleaner and covers the case better than null. Null would require a null default value.

Comment (gfortil) [reply locals.tf:15]: In addition, all three conditions must be true for the entire statement to be true. That dictates whether the admin intends to bring their own virtual network or use the one previously deployed by the virtual network module.

Comment (stuartort) [reply locals.tf:15]: Since all 3 "can" expresssions are evaluated, you can leave this.

I approve this PR

**Review by stuartort — APPROVED** [review]

**Comment by gfortil** [comment]
@richardkchapman please merge
